.github/workflows/push_master.workflow.yml

```diff
@@ -0,0 +1,50 @@
+name: Push checks
+
+on:
+  push:
+    branches:
+      - master
+
+jobs:
+  lint:
+    name: Lint
+    runs-on: ubuntu-18.04
+    steps:
+      - uses: actions/checkout@v2
+      - name: Cache node modules
+        uses: actions/cache@v2
+        env:
+          cache-name: cache-node-modules
+        with:
+          path: ~/.npm
+          key: ${{ runner.os }}-build-${{ env.cache-name }}-${{ hashFiles('**/package-lock.json') }}
+          restore-keys: |
+            ${{ runner.os }}-build-${{ env.cache-name }}-
+            ${{ runner.os }}-build-
+            ${{ runner.os }}-
+      - uses: actions/setup-node@v1.4.4
+        with:
+          node-version: "12"
+      - uses: ./.github/actions/lint
+
+  unit-tests:
+     name: Unit Tests
+     runs-on: ubuntu-18.04
+     needs: [lint]
+     steps:
+       - uses: actions/checkout@v2
+       - name: Cache node modules
+         uses: actions/cache@v2
+         env:
+           cache-name: cache-node-modules
+         with:
+           path: ~/.npm
+           key: ${{ runner.os }}-build-${{ env.cache-name }}-${{ hashFiles('**/package-lock.json') }}
+           restore-keys: |
+             ${{ runner.os }}-build-${{ env.cache-name }}-
+             ${{ runner.os }}-build-
+             ${{ runner.os }}-
+       - uses: actions/setup-node@v1.4.4
+         with:
+           node-version: "12"
+       - uses: ./.github/actions/unit-tests
```

Comment:
Need a job to publish on NPM.js, you can find an example in the Ergol repository (https://github.com/kuzzleio/ergol/blob/8b8793b415831861c21d8c89f7e051d00bb5ef52/.github/workflows/push_master.workflow.yml#L30)